bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/MapUtils.java

```diff
@@ -148,9 +151,59 @@ public static void checkIsMapOnlyOperation(BType mapType, String op) {
         switch (mapType.getTag()) {
             case TypeTags.MAP_TAG:
             case TypeTags.JSON_TAG:
+            case TypeTags.RECORD_TYPE_TAG:
                 return;
             default:
                 throw createOpNotSupportedError(mapType, op);
         }
     }
+
+    public static void checkValidFieldForRecord(MapValue<?, ?> m, String  k, String  op) {
+        BType type = m.getType();
+        switch (type.getTag()) {
+            case TypeTags.RECORD_TYPE_TAG:
+                boolean isValid = checkField(m, k);
+                if (!isValid) {
+                    return;
+                }
+                boolean isRequired = checkForRequiredFields((BRecordType) type, k);
+                if (isRequired) {
+                    throw createOpNotSupportedError(type, op);
+                }
+                return;
+            default:
+                return;
+        }
+    }
+
+    private static boolean checkField(MapValue<?, ?> m, String k) {
+        return m.containsKey(k);
+    }
+
+    private static boolean checkForRequiredFields(BRecordType type, String k) {
+        Map<String, BField> fields = type.getFields();
+        BField field = fields.get(k);
+        if (field != null && Flags.isFlagOn(field.flags, Flags.REQUIRED)) {
+            return true;
+        }
+        return false;
+    }
+
+    public static void validateRecord(BType type, String op) {
+        if (type instanceof BRecordType) {
```

Comment:
Shall we check by tag and also return early if not a record type?

```suggestion
        if (type.tag != TypeTags.RECORD_TYPE_TAG) {
                return;
        }                    
```

Reply:
Will Fix